src/js/BookReader.js

```diff
@@ -2149,99 +2151,67 @@ BookReader.prototype.createPopup = createPopup;
  * Bind navigation handlers
  */
 BookReader.prototype.bindNavigationHandlers = function() {
-  var self = this;
+  const self = this;
 
   // Note the mobile plugin attaches itself to body, so we need to select outside
-  var jIcons = this.$('.BRicon').add('.BRmobileMenu .BRicon');
-
-  jIcons.filter('.onepg').bind('click', function() {
-    self.switchMode(self.constMode1up);
-  });
-
-  jIcons.filter('.twopg').bind('click', function() {
-    self.switchMode(self.constMode2up);
-  });
-
-  jIcons.filter('.thumb').bind('click', function() {
-    self.switchMode(self.constModeThumb);
-  });
+  const jIcons = this.$('.BRicon').add('.BRmobileMenu .BRicon');
+  const navigationControls = {
+    book_left: () => {
+      this.trigger(BookReader.eventNames.stop);
+      this.left();
+    },
+    book_right: () => {
+      this.trigger(BookReader.eventNames.stop);
+      this.right();
+    },
+    book_up: () => {
+      if ($.inArray(this.mode, [this.constMode1up, this.constModeThumb]) >= 0) {
+        this.scrollUp();
+      } else {
+        this.prev();
+      }
+    },
+    book_down: () => {
+      if ($.inArray(this.mode, [this.constMode1up, this.constModeThumb]) >= 0) {
+        this.scrollDown();
+      } else {
+        this.next();
+      }
+    },
+    book_top: this.first,
+    book_bottom: this.last,
+    book_leftmost: this.leftmost,
+    book_rightmost: this.rightmost,
```

Comment:
Will these work? We might need to do `() => this.first()` so that `this` is bound correctly?

Reply:
Tested them out, and they do need to have their context bound. Forgot I'd be in a jQuery context once I called it from within a jQuery callback.